framework/src/auxkernels/BuildArrayVariableAux.C

```diff
@@ -0,0 +1,63 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BuildArrayVariableAux.h"
+
+#include <algorithm>
+
+registerMooseObject("MooseApp", BuildArrayVariableAux);
+
+InputParameters
+BuildArrayVariableAux::validParams()
+{
+  InputParameters params = ArrayAuxKernel::validParams();
+  params.addCoupledVar("component_variables",
+                       "The variables that make up each component of the output array variable.");
+  params.addClassDescription("Combines multiple standard variables into an array variable.");
+  return params;
+}
+
+BuildArrayVariableAux::BuildArrayVariableAux(const InputParameters & parameters)
+  : ArrayAuxKernel(parameters), _component_dofs(coupledAllDofValues("component_variables"))
+{
+  // Check the number of component variables
+  if (_component_dofs.size() != _var.count())
+    paramError("variable",
+               "The array variable has ",
+               _var.count(),
+               " components, but ",
+               _component_dofs.size(),
+               " component variables were specified.");
+
+  // Make sure the FEType of each input variable matches the output type
+  const auto & var_names = parameters.get<std::vector<VariableName>>("component_variables");
+  THREAD_ID tid = parameters.get<THREAD_ID>("_tid");
+  for (VariableName name : var_names)
+  {
+    const MooseVariableFieldBase & var_base = _subproblem.getVariable(tid, name);
+    if (var_base.feType() != _var.feType())
+      paramError(
+          "component_variables",
+          "The input and output variables of BuildArrayVariableAux must have the same FE type");
+  }
+}
+
+void
+BuildArrayVariableAux::compute()
+{
+  unsigned int n_local_dofs = _var.numberOfDofs();
+  _local_sol.resize(n_local_dofs);
+  for (unsigned int j = 0; j < n_local_dofs; ++j)
+  {
+    _local_sol(j) = RealEigenVector::Zero(_var.count());
+    for (unsigned int i = 0; i < _var.count(); ++i)
+      _local_sol(j)(i) = (*_component_dofs[i])[j];
+  }
+  _var.setDofValues(_local_sol);
```

Comment:
```suggestion
  const auto n_local_dofs = _var.numberOfDofs();
  _local_sol.resize(n_local_dofs, RealEigenVector::Zero(_var.count()));
  for (MooseIndex(n_local_dofs) j = 0; j < n_local_dofs; ++j)
    for (MooseIndex(_var.count()) i = 0; i < _var.count(); ++i)
      _local_sol(j)(i) = (*_component_dofs[i])[j];
  }
  _var.setDofValues(_local_sol);
```

Reply:
You didn't fully resolve this

Reply:
`_local_sol` is a `libmesh::DenseVector` and it looks like it doesn't support that kind of `std::vector::resize` syntax.

Reply:
At least do
```
  unsigned int n_local_dofs = _var.numberOfDofs();
  _local_sol.resize(n_local_dofs);
  for (unsigned int j = 0; j < n_local_dofs; ++j)
  {
    _local_sol(j).resize(_var.count());
    for (unsigned int i = 0; i < _var.count(); ++i)
      _local_sol(j)(i) = (*_component_dofs[i])[j];
  }
  _var.setDofValues(_local_sol);
```

No need to allocate a whole new zero vector on each element.